lib/web_ui/test/text_editing_test.dart

```diff
@@ -705,7 +709,7 @@ Future<void> testMain() async {
       expect(spy.messages, hasLength(0));
       await Future<void>.delayed(Duration.zero);
       // DOM element still keeps the focus.
-      expect(defaultTextEditingRoot.activeElement,
+      expect(defaultTextEditingRoot.ownerDocument?.activeElement,
```

Comment:
@eyebrowsoffire this is the change I've made, with the firefox tests passing - however, this is dependent on a few other changes in this PR, mainly `defaultTextEditingRoot` being changed, so it might not be super easy to drop these changes in. 

Oddly enough, I tried running the same tests on the latest version of `main` branch and they still pass with the `--browser=firefox` flag

Reply:
The problem might be with the way Firefox reports `activeElement` when the focus is inside the shadow dom. Now that text editing elements moved out of the shadow dom, the issue is fixed.